base/src/dispatcher/dispatcher.ts

```diff
@@ -579,23 +578,29 @@ export class Dispatcher implements ZLUX.Dispatcher {
         }
         break;
       case ActionType.Minimize:
-          this.log.debug("invoke Launch, which means do nothing if wrapper found: "+wrapper);
-          this.windowManager.minimize(targetId);
-          console.log(targetId);
+          if (targetId) {
+             this.log.info("the targetID is " + targetId);
+             this.windowManager.minimize(targetId);
+
+          }
           break;
       case ActionType.Maximize:
-         this.log.debug("invoke Launch, which means do nothing if wrapper found: "+wrapper);
-         this.windowManager.maximize(targetId);
-        this.log.warn('Max/Min not supported at this time. Concern if apps should be able to control other apps visibility.');
+          if (targetId) {
```

Comment:
You also need to check that this.windowManager exists, and if the whole conditional is not true you should print a `this.log.warn` so that we know why it failed.